src/lib/relayers/BiconomyRelayer.ts

```diff
@@ -0,0 +1,155 @@
+import { Biconomy } from '@biconomy/mexa';
+
+import { SupportedChainId } from '../../constants/chains';
+import {
+    BiconomyRelayerProps,
+    BiconomySendGaslessTransactionParams,
+    BiconomyWalletClientType,
+    InitBiconomyRelayerProps,
+    SendGaslessTransactionType,
+    WebHookAttributesType,
+    ZeroWalletProviderType
+} from '../../types';
+import { delay } from '../../utils/global';
+import { getTransactionReceipt } from '../../utils/provider';
+
+import { BaseRelayer } from './BaseRelayer';
+export class BiconomyRelayer implements BaseRelayer {
+    name = 'Biconomy';
+    chainId: SupportedChainId;
+    #provider: ZeroWalletProviderType;
+    #apiKey: string;
+    #fundingKey: string;
+    #biconomy = {} as any; // eslint-disable-line @typescript-eslint/no-explicit-any
+    #biconomyWalletClient?: BiconomyWalletClientType;
+    #biconomyLoading: Promise<void>;
+
+    constructor(relayerProps: BiconomyRelayerProps) {
+        this.chainId = relayerProps.chainId;
+        this.#provider = relayerProps.provider;
+        this.#apiKey = relayerProps.apiKey;
+        this.#fundingKey = relayerProps.fundingKey;
+
+        this.#biconomyLoading = this.initRelayer({
+            provider: this.#provider
+        } as InitBiconomyRelayerProps);
+    }
+
+    async #waitForBiconomyWalletClient() {
+        await this.#biconomyLoading;
+    }
+
+    async initRelayer(params: InitBiconomyRelayerProps): Promise<void> {
+        this.#biconomy = new Biconomy(params.provider, {
+            apiKey: this.#apiKey
+        });
+
+        const _biconomyWalletClient =
+            await new Promise<BiconomyWalletClientType>((resolve, reject) => {
+                this.#biconomy
+                    .onEvent(this.#biconomy.READY, async () => {
+                        let biconomyWalletClient: BiconomyWalletClientType;
+
+                        try {
+                            do {
+                                biconomyWalletClient =
+                                    this.#biconomy.biconomyWalletClient;
```

Comment:
Isn't there a better way to do this? aren't any hooks exposed?

Reply:
I talked to them about it, they said we shouldn't need to do this. But it was throwing errors on the frontend side so we had to add it.
As far as I see it, there is no other way.